osgp/protocol-adapter-iec61850/osgp-protocol-adapter-iec61850/src/main/java/org/opensmartgridplatform/adapter/protocol/iec61850/infra/networking/services/commands/Iec61850GetStatusCommand.java

```diff
@@ -39,6 +39,12 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(Iec61850GetStatusCommand.class);
 
+    private DeviceMessageLoggingService loggingService;
+
+    public Iec61850GetStatusCommand(final DeviceMessageLoggingService loggingService) {
+        this.loggingService = loggingService;
+    }
+
     public DeviceStatusDto getStatusFromDevice(final Iec61850Client iec61850Client,
             final DeviceConnection deviceConnection, final Ssld ssld) throws ProtocolAdapterException {
         final Function<DeviceStatusDto> function = new Function<DeviceStatusDto>() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this anonymous class number of lines from 50 to at most 20, or make it a named class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1188)